packages/pwa-kit-react-sdk/src/ssr/universal/components/route-component/index.js

```diff
@@ -396,30 +396,28 @@ export const routeComponent = (Wrapped, isPage, locals) => {
 }
 
 /**
+ * TODO: update this comment
  * Wrap all the components found in the application's route config with the
  * route-component HOC so that they all support `getProps` methods server-side
  * and client-side in the same way.
  *
  * @private
  */
-export const getRoutes = (locals = {}) => {
-    let _routes = routes
+export const getAllRoutes = async (locals = {}) => {
     const {applicationExtensions = []} = locals
-    if (typeof routes === 'function') {
-        _routes = routes()
-    }
-
-    // Call the `extendRoutes` function for all the Application Extensions.
-    applicationExtensions.forEach((applicationExtension) => {
-        _routes = applicationExtension.extendRoutes(_routes)
-    })
+    const extensionRoutes = (
+        await Promise.all(applicationExtensions.map((extension) => extension.getRoutes()))
+    ).flat()
```

Comment:
Performance: by replacing `extendRoutes` with the simpler `getRoutes` (no need to pass in the existing/previous routes), we can easily call them in parallel.